### PR TITLE
feat: if profile only has one public event-type, redirect to it

### LINKED ISSUE #12089: [CAL-2648] if profile only has one public event-type, redirect to it
1. https://cal.com/peer only has one visible event-type (others are secret/hidden)
2. visiting https://cal.com/peer should redirect to https://cal.com/peer/meet immediately
3. however, if you click the avatar, it should go back to profile without redirecting (i.e. as a URL parameter: ?redirect=false)
CleanShot 2023-10-25 at 20 36 10@2x

# why:
- having only one event-type means a user can only take one action anyway
- speeds up, so people can send cal.com/peer vs cal.com/peer/meet

<sub>[CAL-2648](https://linear.app/calcom/issue/CAL-2648/if-profile-only-has-one-public-event-type-redirect-to-it)</sub>

### PR DESCRIPTION
**Pull request by ibex088**
Fixes #12089

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## for organizations
[Screencast from 31-10-23 10:52:23 AM IST.webm](https://github.com/calcom/cal.com/assets/32706411/c93c30a9-d96c-40cb-aa21-b245a3867057)
## for non-organizations
[Screencast from 31-10-23 10:54:44 AM IST.webm](https://github.com/calcom/cal.com/assets/32706411/72206dfb-69c5-40be-81d7-20b2f88f6bf9)

### CODE REVIEW

**Review by shvuuuu — APPROVED** [review]
Fixed it pretty well

**Review by Udit-takkar — COMMENTED** [review]
Code looks fine to me

File: apps/web/components/ui/avatar/UserAvatarGroup.tsx [hunk 1]
```diff
@@ -11,6 +12,7 @@ export function UserAvatarGroup(props: UserAvatarProps) {
     <AvatarGroup
       {...rest}
       items={users.map((user) => ({
+        href: `${CAL_URL}/${user.username}?redirect=false`,
```
Comment (PeerRich) [inline apps/web/components/ui/avatar/UserAvatarGroup.tsx:15]: bear in mind this is likely used in other places, no?

**Comment by PeerRich** [comment]
great work!

**Review by PeerRich — APPROVED** [review]

File: apps/web/pages/[user].tsx [hunk 2]
```diff
@@ -388,6 +389,16 @@ export const getServerSideProps: GetServerSideProps<UserPageProps> = async (cont
     descriptionAsSafeHTML: markdownToSafeHTML(eventType.description),
   }));
 
+  // if profile only has one public event-type, redirect to it
+  if (eventTypes.length === 1 && context.query.redirect !== "false") {
+    return {
```
Comment (zomars) [inline apps/web/pages/[user].tsx:394]: I think this is causing dynamic links to not work properly when users only have 1 event type.

**Pull request merged**